src/app/Main/Body/QueryPage/Drawer/Displayed_Features/DisplFeatures-Selector.component.tsx

```diff
@@ -74,7 +74,7 @@ export function DisplayedFeatureSelectorComponent(
     const handleChange = (
         selectedOption: ValueType<{ value: string; label: string }>,
         keyName: FilterType | TableType
-    ): void => props.onChange(selectedOption, keyName);
+    ): void => props.onChange(selectedOption as { value: string; label: string }, keyName);

```

Comment:
reject: see comment above: you should do type checking here, something like
```
if(selectedOption !== undefined && !Array.isArray(selectedOption) {
  props.onChange(...)
} else {
  //  depends on what selection is allowed and is a reasonable output for this component
  throw error or just do nothing or
  props.onChange({value: "", label: ""}) or
  props.onChange(null)
}
```
So you check if the ValueType looks like you expect it to look like.

Reply:
Done. Commit "Type validation in selector functions"